pkg/didcomm/transport/http/outbound_test.go

```diff
@@ -0,0 +1,179 @@
+/*
+	Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+	SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+const certPrefix = "../../../../test/fixtures/keys/"
+
+func TestWithOutboundOpts(t *testing.T) {
+	opt := WithOutboundHTTPClient(nil)
+	require.NotNil(t, opt)
+	clOpts := &outboundCommHTTPOpts{}
+	opt(clOpts)
+
+	opt = WithOutboundTimeout(10 * time.Second)
+	require.NotNil(t, opt)
+	clOpts = &outboundCommHTTPOpts{}
+	// opt.client is nil, so setting timeout should panic
+	require.Panics(t, func() { opt(clOpts) })
+
+	opt = WithOutboundTLSConfig(nil)
+	require.NotNil(t, opt)
+	clOpts = &outboundCommHTTPOpts{}
+	opt(clOpts)
+}
+
+func TestOutboundHTTPTransport(t *testing.T) {
+	// prepare http l
+	l := startMockServer()
```

Comment:
@Baha-sk let's rename this variable to `server` for readability

Reply:
done